ui/src/app/i18n-strings.ts

```diff
@@ -0,0 +1,31 @@
+import {InjectionToken} from '@angular/core';
+
+import {LanguageCode, StringAssets} from './i18n-helper';
+
+// This type define makes sure the language code used in ALL_LANGUAGE_ASSETS is
+// expected and no missing field in any translation.
+type LanguageAssets = Partial<Record<LanguageCode, StringAssets>>;
```

Comment:
Is this 'Partial' needed? IIUC it makes the key and value of the Record optional, which isn't really what we want here.

Reply:
It is actually what we want. Otherwise, we would have to include all LanguageCode as key in ALL_LANGUAGE_ASSETS. And when a key presents in ALL_LANGUAGE_ASSETS, our code assumes the corresponding languages is supported.

We want each organization has the ability to optionally add/remove any language.

Reply:
Ah, thanks! I looked up the type definition for Record and it made more sense.